### PR TITLE
Possible fix for keyboard crash

### PR DESCRIPTION
**Pull request by gcalero**
Testing a possible fix for keyboard crash in android

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8029/app_12643_PR.apk

File: libraries/qml/src/qml/OffscreenSurface.cpp [hunk 1]
```diff
@@ -181,7 +181,17 @@ bool OffscreenSurface::eventFilter(QObject* originalDestination, QEvent* event)
...
+            auto window = getWindow();
+            if (window && window->activeFocusItem()) {
+                event->ignore();
+                if (QCoreApplication::sendEvent(window->activeFocusItem(), event)) {
+                    return event->isAccepted();
+                }
+                return false;
+            }
+            break;
+        }
```
Comment (SamGondelman) [inline libraries/qml/src/qml/OffscreenSurface.cpp]: can we do:
```
        case QEvent::InputMethod:
        case QEvent::InputMethodQuery: {
            auto window = getWindow();
            if (window && window->activeFocusItem()) {
                event->ignore();
                if (QCoreApplication::sendEvent(window->activeFocusItem(), event)) {
                    if (event-type() == QEvent::InputMethodQuery) {
                        QInputMethodQueryEvent* imqEvent = static_cast<QInputMethodQueryEvent*>(event);
                        // this block disables the selection cursor in android which appears in
                        // the top-left corner of the screen
                        if (imqEvent->queries() & Qt::ImEnabled) {
                            imqEvent->setValue(Qt::ImEnabled, QVariant(false));
                        }
                    }
                    return  event->isAccepted();
                }
                return false;
            }
            break;
        }
```

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8029/HighFidelity-Beta-PR12643-e744edc6a0532e5d9738af066ad2d622f6d958db.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8029/HighFidelity-Beta-PR12643-e744edc6a0532e5d9738af066ad2d622f6d958db.exe

File: libraries/qml/src/qml/OffscreenSurface.cpp (hunk 1 above)
Comment (gcalero) [reply libraries/qml/src/qml/OffscreenSurface.cpp]: sure

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8035/app_12643_PR.apk

**Comment by SamGondelman** [comment]
This fixed the crash on both the pixel 2 and the mate 9 pro, so I'll approve and merge once the changes are made 😄

**Comment by SamGondelman** [comment]
Tested on pixel 2 and mate 9 pro 👍

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8035/HighFidelity-Beta-PR12643-d0f1aa9020736917ee3baf647a1c530dd5b3986d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8035/HighFidelity-Beta-PR12643-d0f1aa9020736917ee3baf647a1c530dd5b3986d.dmg

**Pull request merged**